src/playerclass/src/class_item_restriction.c

```diff
@@ -1,22 +1,57 @@
 #include <assert.h>
 #include <stdbool.h>
 #include <stdlib.h>
+#include <string.h>
 #include "game-state/item.h"
 #include "playerclass/class.h"
 #include "playerclass/class_structs.h"
 
 /* see class_item_restriction.h */
-int set_item_restriction(item_t* item, class_t* class) {
+int add_item_restriction(item_t* item, class_t* class) {
     if ((item != NULL) && (class != NULL)) {
-        set_bool_attr(item, class->name, true);
-        return SUCCESS;
+        if (item->class_restrictions == NULL) {
+            item->class_restrictions = create_list_attribute();
+        }
+
+        /* Attribute creation will be refactored in game-state/item to allow for streamlined attribute creation */
+        attribute_t* restriction = malloc(sizeof(attribute_t));
+        restriction->attribute_tag = BOOLE;
+        restriction->attribute_value.bool_val = true;
+        restriction->attribute_key = strndup(class->name, 100);
+
+        if (restriction == NULL) {
+            fprintf(stderr, "Failed to allocate memory");
```

Comment:
Two things here: 
1. I think you should return immediately if this error occurs.  We probably do not want to add NULL to `item->class_restrictions`.
2. I think it is recommended to add the location to an error print statement, to make it bit easier to pin down the bug if it does occur.  So like:
```suggestion
            fprintf(stderr, "Failed to allocate memory in class_item_restriction");
```
This was at least the custom adopted by whoever wrote the code I worked on (class.c).  It seems like a good idea.  You should do this everytime you `fprintf` an error message.

Reply:
Thank you for pointing this out!